packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -346,7 +350,14 @@ export abstract class Chart extends Observable implements AgChartInstance {
         this.animationManager.skip();
         this.animationManager.play();
 
-        this.processors = [new BaseLayoutProcessor(this, this.layoutService)];
+        this.dataLazyLoader = new DataLazyLoader<any>();
+
+        this.processors = [
+            new BaseLayoutProcessor(this, this.layoutService),
+            new DataWindowProcessor(this, this.dataLazyLoader, this.updateService, this.zoomManager, (data) => {
+                this.data = data;
+            }),
+        ];
```

Comment:
I like extracting this out of `Chart`, but this data setting callback is a little meh. On the other hand, a data service to change it seems too much?